compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIETemporaryAllocBufferization.cpp

```diff
@@ -13,47 +13,42 @@ namespace mlir::iree_compiler::AMDAIE {
 
 namespace {
 
-static std::optional<BufferOp> createBufferForTemporaryAllocOp(
-    IRRewriter &rewriter, WorkgroupOp workgroupOp, memref::AllocOp allocOp,
-    CoreOp coreOp, unsigned index) {
-  OpBuilder::InsertionGuard g(rewriter);
-  TileOp tileOp = coreOp.getTileOp();
-  // Reset rewriter's location to after last tile's declaration.
-  auto tiles = workgroupOp.getBody()->getOps<TileOp>();
-  assert(!tiles.empty() && "no tiles in workgroupOp");
-  rewriter.setInsertionPointAfter(*std::prev(tiles.end(), 1));
-  auto bufferType = cast<MemRefType>(allocOp.getType());
-  auto bufferOp = rewriter.create<AMDAIE::BufferOp>(
-      rewriter.getUnknownLoc(), bufferType, tileOp, nullptr);
-  return bufferOp;
-}
-
-static LogicalResult bufferizeTemporaryAllocInCoreOp(
-    IRRewriter &rewriter, WorkgroupOp workgroupOp, CoreOp coreOp,
-    SmallVector<Operation *> &toBeErased) {
-  // Step 1. Get all buffers within a CoreOp.
-  SmallVector<memref::AllocOp> allocOps;
-  coreOp.walk([&](Operation *op) {
-    if (auto allocOp = dyn_cast<memref::AllocOp>(op)) {
-      allocOps.push_back(allocOp);
-      toBeErased.push_back(allocOp);
-    } else if (auto deallocOp = dyn_cast<memref::DeallocOp>(op)) {
-      toBeErased.push_back(deallocOp);
+LogicalResult bufferizeTemporaryMemrefs(Operation *parentOp) {
```

Comment:
While this is an improvement, I think we should go from `access` directly to `amdaie.buffer` and avoid the alloc intermediary.

Reply:
Yes ok I think that makes sense, I'll pick that up next.